generator/src/main/java/org/apache/kafka/message/JsonConverterGenerator.java

```diff
@@ -375,13 +385,36 @@ private void generateVariableLengthTargetToJson(Target target, Versions versions
                         target.sourceVariable())));
             } else {
                 headerGenerator.addImport(MessageGenerator.ARRAYS_CLASS);
+                headerGenerator.addImport(MessageGenerator.INT_NODE_CLASS);
+                buffer.printf("if (_serializeRecords) {%n");
```

Comment:
Yea, this change is because ProduceRequest uses `bytes` type. It would be worthwhile if we waited until we migrated the produce request to the automated protocol so we wouldn't have to change this.

Reply:
Understood. I think that we should revert this. I think that it makes sense to wait until we complete the migration of the remaining requests. We should have them pretty soon now.